src/Services/Catalog/Catalog.Domain/Abstractions/ICategoryRepository.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Catalog.Domain.Entities;
+
+namespace Catalog.Domain.Abstractions;
+
+public interface ICategoryRepository
+{
+    Task<string> CreateCategoryAsync(Category category, CancellationToken cancellationToken);
```

Comment:
you should define IDaseRepository interface where you will describe all of basic CRUD methods

Reply:
so I base repository has to be generic?